### ISSUE TITLE
Install specific version in centos

### ISSUE THREAD

**Issue by giovannicandido** [haxorof/ansible-role-docker-ce#31]
Hi,
To create a kubernets clusters is necessary the installation of docker 17.03 which is the last supported.

Trying to change the name of package gives the error:

```
changed": true, "msg": "Error: Package: docker-ce-17.03.2.ce-1.el7.centos.x86_64 (docker-ce-st
able)\n           Requires: docker-ce-selinux >= 17.03.2.ce-1.el7.centos\n           Available: docker-ce-selinux-17.03.0.ce-
1.el7.centos.noarch (docker-ce-stable)\n               docker-ce-selinux = 17.03.0.ce-1.el7.centos\n 

```

Which says package docker-ce-selinux is incompatible.

To solve the problem it needs install both at the same time:

```
yum install -y --setopt=obsoletes=0
docker-ce-17.03.0.ce-1.el7.centos
docker-ce-selinux-17.03.0.ce-1.el7.centos
```

**Comment by haxorof** [comment]
Thanks @giovannicandido for reporting this. Will have a look at this and see what can be done in the role to best solve this.

Cheers!

**Comment by haxorof** [comment]
Hi,
I have looked into this a bit and it seems related to the "obsoletes" option specified in yum and Docker CE packages. When you install this older Docker CE 17.03 package it somehow fails because obsolete restrictions was added in Docker CE 17.06 package. Does it work for you @giovannicandido if you install Docker CE like this:
```bash
yum install --setopt=obsoletes=0 docker-ce-17.03.0.ce-1.el7.centos
```

If this works for you then one option would be to globally set `obsoletes` to `0` in `/etc/yum.conf`. It might not be that easy to fix in this role just for this version of Docker because if you try 17.06 and higher it works as expected.

Please let me know if the `obsoletes` resolves you installation issues related to 17.03.

Cheers!

**Comment by giovannicandido** [comment]
It works, attention to the caveat of conflict if downgrading is necessary the remove of dependencies

```
yum remove docker-common container-selinux docker-selinux docker-engine
```

Also to make sure the config is on CentOS here is a ansible reciple:

```
- name: ensure old versions of docker can be installed in CentOS
  lineinfile:
    path: /etc/yum.conf
    regexp: 'obsoletes=\d'
    line: 'obsoletes=0'
  when: ansible_distribution == 'CentOS'
```
